File: php/features.php
<?php
declare(strict_types=1);

// WhoisDomainMonitoring SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WhoisDomainMonitoringFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WhoisDomainMonitoringBaseFeature();
            case "test":
                return new WhoisDomainMonitoringTestFeature();
            default:
                return new WhoisDomainMonitoringBaseFeature();
        }
    }
}

File: py/features.py
# WhoisDomainMonitoring SDK feature factory

from feature.base_feature import WhoisDomainMonitoringBaseFeature
from feature.test_feature import WhoisDomainMonitoringTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WhoisDomainMonitoringBaseFeature(),
        "test": lambda: WhoisDomainMonitoringTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
